sniffles/sniffles.py

```diff
@@ -177,22 +186,36 @@ def start_generation(sconf):
         flow_start_offset = random.randint(
             1, sconf.getConcurrentFlows() + 100000
         )
-        # !!Create background traffic conversation based on
+        # Create background traffic conversation based on
         # Background traffic rule
         if back_traffic_percent > 0:
             pick = random.randint(0, 99)
             if pick < back_traffic_percent:
                 btrule = Rule("Background Traffic")
-                btr_ts = BackgroundTrafficRule()
+                # Pick protocol if distribution is known
+                if back_traffic_distribution:
+                    protocol_pick = random.randint(0, 99)
+                    protocol = None
+                    pick_sum = 0
+                    for proto in back_traffic_distribution:
+                        pick_sum += back_traffic_distribution[proto]
+                        if (protocol_pick < pick_sum):
+                            protocol = proto
+                            break
+                    if protocol is None:
```

Comment:
If we have a distribution, why would we ever arrive at a point where we don't get a protocol?